hicommonwealth/commonwealth
# Added external token integration functionality in community integrations

## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/9898

## Description of Changes
Added option to pin tokens to the community. Users can pin any token that is launched externally.

## "How We Fixed It"
N/A

## Test Plan
Pre-req: `pnpm migrate-db`

1. Visit the community integrations page of any community that you are an admin of.
    1. If this community is created on a non-Ethereum network, verify you don't see the "Add token" section in the integration page and visiting `/:communityId/manage/integrations/token` shows a "Page not found"
    2. If this community is created on an Ethereum network
        1. If this community already has a launchpad token (a token that created this community), verify you don't see the "Add token" section in the integration page and visiting `/:communityId/manage/integrations/token` shows a "Page not found"
        2. If this community doesn't have a launchpad token
                  1. Verify you see the "Add token" section on the integration page if there is not already a token connected to the community
                  2. Verify you see the "Manage connected token" section on the integration page if there is already a token connected to the community
                  3. Verify visiting `/:communityId/manage/integrations/token` doesn't show a "Page not found"
2. For `1.2.2.1`, Click on the "Add token" button, fill the form and add a token ex: add shushi with `0x7d49a065d17d6d4a55dc13649901fdbb98b2afba` address. 
    1. Verify you are able to add this token correctly.
    2. Verify that you see "Token connected" status and the UI updates to reflect the connected token info
    3. Verify that on the `/:communityId/manage/integrations` page, the "Add token" section now becomes "Manage connected token" with status indicating that a token is connected to community.
3. For `1.2.2.2`, Click on the "Manage connected token" button
    1. Verify you are able to "remove" this connected token, and the UI updates to reflect "disconnected" status
    2. Verify you are able to "change" this connected token, and the UI updates to reflect "disconnected" status
        1. Verify that if you attempt to add the same token again, it show info message indicating that this token is already connected.


## Deployment Plan
N/A

## Other Considerations
There are some followup tickets that would add related functionality which is not ready to add atm
- https://github.com/hicommonwealth/commonwealth/issues/10233
- https://github.com/hicommonwealth/commonwealth/issues/10234
- https://github.com/hicommonwealth/commonwealth/issues/10231
